woocommerce/woocommerce
# POC: Add plugin integration scripts

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:

This PR adds scripts to assist plugins integrate with `@wordpress/scripts` and make the blocks build process easier.  This is heavily inspired by `@wordpress/create-block` but attempts to remove singular block creation from the general integration process.

<img width="834" alt="Screen Shot 2023-09-01 at 4 43 10 PM" src="https://github.com/woocommerce/woocommerce/assets/10561050/65549774-559d-495c-af11-491d5c7256fd">
<img width="1314" alt="Screen Shot 2023-08-31 at 5 18 44 PM" src="https://github.com/woocommerce/woocommerce/assets/10561050/4a90e2ed-5d5f-46e7-9122-043f5df069c0">


Closes # .



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Check out this branch and `pnpm i`
2. Navigate to the `woocommerce/packages/js/itegrate-plugin` directory
3. Run `npm link`
4. In a test plugin directory, run `npx woo-integrate-plugin`, optionally with `--wp-env` and/or `--wp-scripts` arguments.  `npx woo-integrate-plugin --wp-env --wp-scripts`
5. Check that the following occur:

* [ ] A package.json file is created if one does not exist, using the name and text domain from the plugin's entry file (plugin metadata)
* [ ] A composer.json file is created if one does not exist, using the name and text domain from the plugin's entry file (plugin metadata)
* [ ] `--wp-scripts` - `@wordpress/scripts` is installed as a dev dependency
* [ ] `--wp-scripts` - Scripts are added to the package.json file to build, start, lint, etc if the script key has not already been used
* [ ] `--wp-scripts` - A warning is shown if a script cannot be added and the command does not contain the expected script command
* [ ] `--wp-scripts` - If a webpack config exists, an error should be shown if `@wordpress/scripts` is not being used in this file
* [ ] `--wp-env` - `@wordpress/env` is installed as a dev dependency
* [ ] `--wp-env` - A `.wp-env` file is created in the plugin root directory

6. Test with the template `npx woo-integrate-plugin --template=/{FULL_PATH}/woocommerce/packages/js/product-editor-extension`
7. Make sure that the template files are copied over with the correct namespaces
8. Check that the `.woo-plugin.json` file is created with the `BlockRegistry` module.
9. Make sure that the instructions to add lines to your plugin is shown at the end of the integration process
10. Add the lines to the plugin and make sure that the block shows on the add product page



### Changelog entry




<details>

#### Significance




#### Type




#### Message 

#### Comment 

</details>
